axon-framework/axon-framework-commands/infrastructure.md

```diff
@@ -414,59 +376,242 @@ public DisruptorCommandBus commandBus(TransactionManager txManager, AxonConfigur
 
-* [SpringCloud](../../extensions/spring-cloud.md) or 
-* [JGroups](../../extensions/jgroups.md).
+The `AnnotationRoutingStrategy`  and `MetaDataRoutingStrategy` are considered the full implementations to configure.
+The `ERROR`, `RANDOM_KEY` and `STATIC_KEY` are _fallback routing strategies_ that should be configured on the annotation or meta-data implementations.
+To get a grasp how these are constructed, consider the following sample:
 
-Configuring a distributed command bus can \(mostly\) be done without any modifications in configuration files.
+{% tabs %}
+{% tab title="AnnotationRoutingStrategy" %}
+```java
+// A custom annotation can be used to drive the AnnotationRoutingStrategy
+@interface CustomRoutingAnnotation {
+}
+// Constructs a AnnotationRoutingStrategy with a different annotation and fallback: 
+public RoutingStrategy routingStrategy() {
+    return AnnotationRoutingStrategy.builder()
+                                    .annotationType(CustomRoutingAnnotation.class)
+                                    .fallbackRoutingStrategy(UnresolvedRoutingKeyPolicy.STATIC_KEY)
+                                    .build();
+}
+```
+{% endtab %}
+
+{% tab title="MetaDataRoutingStrategy" %}
+```java
+// Constructs a MetaDataRoutingStrategy with the defined meta data key and a different fallback:
+public RoutingStrategy routingStrategy() {
+    return MetaDataRoutingStrategy.builder()
+                                  .metaDataKey("my-routing-key")
+                                  .fallbackRoutingStrategy(UnresolvedRoutingKeyPolicy.RANDOM_KEY)
+                                  .build();
+}
+```
+{% endtab %}
+{% endtabs %}
+
+Of course a custom implementation of the `RoutingStrategy` can also be provided when necessary
+When we need to deviate from the default `AnnotationRoutingStrategy`, we should configure it like so: 
+
+{% tabs %}
+{% tab title="Axon Configuration API" %}
+```java
+public class AxonConfig {
+    // ...  
+    public void configureRoutingStrategy(Configurer configurer, RoutingStrategy routingStrategy) {
+        configurer.registerComponent(RoutingStrategy.class, config -> routingStrategy);
+    }
+}
+```
+{% endtab %}
 
-First of all, the starters for one of the Axon distributed command bus modules needs to be included \(e.g. [JGroups](../../extensions/jgroups.md) or [Spring Cloud](../../extensions/spring-cloud.md)\).
+{% tab title="Spring Boot AutoConfiguration" %}
+```java
+@Configuration
+public class AxonConfig {
+    // ...
+    @Bean
+    public RoutingStrategy routingStrategy() {
+        return /* construct your routing strategy */;
+    }
+}
+```
+{% endtab %}
+{% endtabs %}
+
+### AxonServerCommandBus
+
+The `AxonServerCommandBus` is the _default_ distributed `CommandBus` implementation set by the framework.
+It connects to [AxonServer](../../axon-server-introduction.md) to submit and receive commands with.
```

Comment:
```suggestion
It connects to [AxonServer](../../axon-server-introduction.md), with which it can submit and receive commands.
```

Reply:
I would change `submit` by `send`... never heard `submit a command` before 😂

Reply:
Both comments (or commands :thinking: ) are correct. Doing both and changing the sentence to:

`It connects to [AxonServer](../../axon-server-introduction.md), with which it can send and receive commands.`